src/modules/shared/pages/switch-context/switch-context.component.ts

```diff
@@ -1,112 +1,105 @@
+
+        if (currentUserContext) {
+          label = currentUserContext.roleId === role.id ? `Continue as ${label}` : `Switch to ${label} profile`;
+        }
+
+        return {
+          id: role.id,
+          label,
+          type: role.role,
+          ...role.organisation && { organisation: role.organisation },
           ...role.organisationUnit && { organisationUnit: role.organisationUnit }
-          } }
+        };
+
       })
-    });
 
-    this.title = this.initialSelection ? 'Choose your profile' : 'Switch profile';
+    };
+
+    this.setPageTitle(!currentUserContext ? 'Choose your profile' : 'Switch profile');
+
   }
 
-  redirectToHomepage(role: {
-    profile: string,
-    roleId: string,
-    type: UserRoleEnum,
-    organisation?: {
-      id: string,
-      name: string,
-      acronym: string | null,
-      organisationUnit?: { id: string; name: string; acronym: string; }
-    },
-  }): void {
-
-    if (this.currentUserProfile !== role.profile) {
-      let message = '';
-      const currentUserContext = this.authenticationStore.getUserContextInfo();
-
-      const roleName = role.organisation?.organisationUnit ? 
-        `${this.authenticationStore.getRoleDescription(role.type).trimEnd().toLowerCase()} (${role.organisation.organisationUnit.name.trimEnd()})` :
-        `${this.authenticationStore.getRoleDescription(role.type).trimEnd().toLowerCase()}`;
-
-      this.authenticationStore.updateSelectedUserContext(role);
-
-      message = (currentUserContext?.type === role.type && currentUserContext?.organisation?.organisationUnit?.id === role.organisation?.organisationUnit?.id) ? 
-        `You are logged in as ${UtilsHelper.indefiniteArticle(roleName)}.`:
-        `Switch successful: you are now logged in with your ${roleName} profile.`;
-
-      LocalStorageHelper.setObjectItem("role", role);
-  
-      if (!this.initialSelection) {
-        this.setRedirectAlertSuccess(message);
+
+  onSubmit(chosenRoleId: string): void {
+
+    const role = this.user.roles.find(item => item.id === chosenRoleId);
+
+    if (!role) {
+      console.error('Chosen role not found!');
+      return;
+    }
+
+    if (this.currentRole?.id !== role.id) {
+
+      this.authenticationStore.setUserContext({
+        id: this.user.id,
+        roleId: role.id,
+        type: role.type,
+        organisation: role.organisation,
+        organisationUnit: role.organisationUnit
+      });
+
+      if (!this.currentRole) {
+
+        const roleDescription = `${this.authenticationStore.getRoleDescription(role.type).toLowerCase()}${role.organisationUnit ? ` (${role.organisationUnit.name})` : ''}`;
+        this.setRedirectAlertSuccess(`You are now logged in as ${roleDescription}.`);
       }
-    }   
+
+    }
 
     this.redirectTo(`${this.authenticationStore.userUrlBasePath()}/dashboard`);
-  }
 
-  currentlyLoggedMessage() {
-    return `You are currently logged in as ${UtilsHelper.indefiniteArticle(this.currentUserProfile)}`;
```

Comment:
Did you keep the _**You are currently logged in as**_  message anywhere? its part of the flow if the user selects the previous role

Reply:
Yes. Moved it to the HTML.

Make sense?

Reply:
yes